### PR TITLE
Use rounded close time in Consensus (RIPD-1528)

### PR DESCRIPTION
**Pull request by bachase**
Switches the default behavior of Consensus to use `roundCloseTime` instead of `effCloseTime`. `effCloseTime` is still used when accepting the consensus ledger to ensure the consensus close time comes after the parent ledger close time. This change eliminates an edge case in which peers could reach agreement on the close time, but end up generating ledgers with different close times.

### CODE REVIEW

File: src/test/consensus/Consensus_test.cpp [hunk 1]
```diff
@@ -522,6 +522,162 @@ class Consensus_test : public beast::unit_test::suite
...
+            }
+
+
+            // All peers submit their own ID as a transaction
+            for (Peer & peer : sim.peers)
+                peer.submit(Tx{static_cast<std::uint32_t>(peer.id)});
+
+            // Run 1 more round, this time it will have a decreased
+            // resolution of 20 seconds.
+            // 86481
```
Comment (mellery451) [inline src/test/consensus/Consensus_test.cpp]: although I see this value mentioned as raw close time in the following comments, I don't think it has any context in this particular comment  - maybe remove?

File: src/test/consensus/Consensus_test.cpp [hunk 2]
```diff
@@ -522,6 +522,162 @@ class Consensus_test : public beast::unit_test::suite
...
+                {
+                    BEAST_EXPECT(
+                        peer.lastClosedLedger.id() ==
+                        sim.peers[0].lastClosedLedger.id());
+                }
+            }
+            else
+            {
+                // All fast peers agreed on LCL TX id
+                BEAST_EXPECT(std::all_of(
```
Comment (mellery451) [inline src/test/consensus/Consensus_test.cpp]: isn't this check effectively the same as the one starting on line 635, so code could be DRY'd?

Comment (bachase) [reply src/test/consensus/Consensus_test.cpp]: Good catch, I'll fix. For context, these changes were actually back ported from a branch based off of #2209 where the comparisons need to be different.

**Comment by nbougalis** [comment]
Merged as c76656cf7f02d8d848c31b96ec3dc9de3727aaaa

**Pull request closed without merging**